sale_restricted_qty/views/sale_views.xml

```diff
@@ -7,81 +7,42 @@
-                    class="o_form_label"
-                    colspan="2"
-                    attrs="{'invisible': ['|',  ('is_qty_less_min_qty','=', False),
-                     '&amp;', ('is_qty_less_min_qty','=', True), ('force_sale_min_qty','=', False,)]}"
-                    style="background-color:yellow"
-                >Min Quantity Recommended</span>
-                 <span
-                    class="o_form_label"
-                    colspan="2"
-                    attrs="{'invisible': ['|',  ('is_qty_less_min_qty','=', False),
-                     '&amp;', ('is_qty_less_min_qty','=', True), ('force_sale_min_qty','=', True,)]}"
-                    style="background-color:red"
-                >Min Quantity Required</span>
-                 <span
-                    class="o_form_label"
-                    colspan="2"
-                    attrs="{'invisible': [('is_qty_not_multiple_qty','=', False)]}"
-                    style="background-color:red"
-                >Multiple Quantity Required</span>
-                 <span
-                    class="o_form_label"
-                    colspan="2"
-                    attrs="{'invisible': ['|',  ('is_qty_bigger_max_qty','=', False),
-                     '&amp;', ('is_qty_bigger_max_qty','=', True), ('force_sale_max_qty','=', False,)]}"
-                    style="background-color:yellow"
-                >Max Quantity Recommended</span>
-                 <span
-                    class="o_form_label"
-                    colspan="2"
-                    attrs="{'invisible': ['|',  ('is_qty_bigger_max_qty','=', True),
-                     '&amp;', ('is_qty_bigger_max_qty','=', True), ('force_sale_max_qty','=', True,)]}"
-                    style="background-color:red"
-                >Max Quantity Exceeded</span>
-           </xpath>
+            </xpath>
+            <xpath
+                expr="//field[@name='order_line']/form//field[@name='product_uom_readonly']"
+                position="before"
+            >
+                <div style="color:red; font-weight:bold;">
+                    <field
+                        name="display_warning_message_qty"
+                        nolabel="1"
+                        readonly="1"
+                    />
+                </div>
+            </xpath>
            <xpath
                 expr="//field[@name='order_line']/tree/field[@name='product_uom_qty']"
                 position="after"
             >
                <field name="sale_min_qty" />
-               <field name="is_qty_less_min_qty" invisible="1" />
-               <field name="force_sale_min_qty" invisible="1" />
                <field name="sale_max_qty" />
-               <field name="is_qty_bigger_max_qty" invisible="1" />
-               <field name="force_sale_max_qty" invisible="1" />
-               <field name="is_qty_not_multiple_qty" invisible="1" />
+               <field name="display_option_ids_danger" invisible="1" />
```

Comment:
last thing to fix here:
use danger if "qty_invalid"
use warning if "qty_invadid == False and qty_warning_message"